test/unit/callbacks_and_delays_test.lua

```diff
@@ -0,0 +1,86 @@
+local expirationd = require("expirationd")
+local fiber = require("fiber")
+local t = require("luatest")
+
+local helpers = require("test.helper")
+
+local g = t.group('callbacks_and_delays', {
+    {index_type = 'TREE', engine = 'vinyl'},
+    {index_type = 'TREE', engine = 'memtx'},
+    {index_type = 'HASH', engine = 'memtx'},
+})
+
+g.before_each({index_type = 'TREE'}, function(cg)
+    t.skip_if(cg.params.engine == 'vinyl' and helpers.vinyl_is_broken(),
+            'Blocked by https://github.com/tarantool/tarantool/issues/6448')
+    g.space = helpers.create_space_with_tree_index(cg.params.engine)
+end)
+
+g.before_each({index_type = 'HASH'}, function(cg)
+    g.space = helpers.create_space_with_hash_index(cg.params.engine)
+end)
+
+g.before_each({index_type = 'BITSET'}, function(cg)
+    g.space = helpers.create_space_with_bitset_index(cg.params.engine)
+end)
+
+g.after_each(function(g)
+    g.space:drop()
+end)
+
+function g.test_delays_and_scan_callbacks(cg)
+    local space = cg.space
+    local total = 10
+    for i = 1, total do
+        space:insert({i, tostring(i)})
+    end
```

Comment:
It is actually a test setup and can be moved to `g.before_test`.

Reply:
fixed